frontend/frame/EmbedContentFactory.ts

```diff
@@ -0,0 +1,19 @@
+import { ContentFactory } from './ContentFactory';
+import { EmbedContent } from '../../model/slide/content/EmbedContent';
+
+/**
+ * Class which transforms JSON to EmbedContent Object
+ */
+export class EmbedContentFactory extends ContentFactory {
+  /**
+   * Factory-Object,transforms JSON to EmbedContent Object
+   * @param json the JSON sent by the Webserver
+   * @returns the buildContent as BuildContent type
+   * @protected
+   */
+  public buildContent(json: unknown): EmbedContent {
+    const embedContent = new EmbedContent();
+    embedContent.url = (json as EmbedContent).url;
+    return embedContent;
```

Comment:
they all need there type set too.
The best thing would be if you added a constructor for all of them that sets their own type wo the coresponding Value